doc/src/base/collections.md

```diff
@@ -235,6 +236,49 @@ Partially implemented by:
   * [`Iterators.Pairs`](@ref)
 
 ## Set-Like Collections
+A [`Set`](@ref) is a collection of elements with no duplicated elements and keys/indices.
+
+So in contrast to `Arrays`, in a `Set` you can have only one of each element, and the order of elements isn't important (whereas an array can have multiple copies of an element and their order is remembered). As a result, iterating over `Set` elements is slower than `Arrays`.
+
+```jldoctest set_collections
+julia> colors = Set(["yellow", "blue", "green", "red"])
+Set(["yellow", "blue", "green", "red"])
+```
+
+You can use [`push!`](@ref) to add elements to a set:
+```jldoctest set_collections
+julia> push!(colors, "black")
+Set(["yellow", "blue", "green", "black", "red"])
+```
+
+If you try to add something to the set that's already there nothing will happen, because sets don't store repeated elements.
+
+You can't use [`pushfirst!`](@ref), because that works only for things that have a concept of "first", like arrays.
+
+To see if something is in the set, you can use [`in`](@ref):
+```jldoctest set_collections
+julia> "green" in colors  # or: in("green", colors)
+true
+```
+
+To access each element of a `Set` you can loop over the elements:
+```jldoctest set_collections
+julia> for color in colors
+  println(color)
+end
+yellow
+blue
+green
+red
+```
+You can also [`filter`](@ref) some of the elements:
```

Comment:
Quite a lot of "You can", "You can", You can't" in this section.

Reply:
In general, I am not sure if giving examples of many of the functions on `Set` are so useful here since it duplicated the ones in the docstrings. If anything, something like

> common functions used on a `Set` are `in`[@ref], `filter`[@ref] ...

could be done, and then the examples are in those docstrings.

Reply:
> 
> 
> In general, I am not sure if giving examples of many of the functions on `Set` are so useful here since it duplicated the ones in the docstrings. If anything, something like
> 
> > common functions used on a `Set` are `in`[@ref], `filter`[@ref] ...
> 
> could be done, and then the examples are in those docstrings.

These functions are not given in the following docstrings, and so I thought they might be useful to have them here.